python/raf/distributed/op.py

```diff
@@ -151,6 +151,8 @@ def reduce_scatter(x, computation="sum", rank_list=None):
         reduction result of x[rank] over all replicas,
         where rank represents rank number of the current process
     """
+    if isinstance(x, (tuple, list)):
```

Comment:
add checks here:
- tensors in a tuple/list should have the same size
- if `rank_list` is None, `len(x)` should be global comm's size
- if `rank_list`, parse the group info, `len(x)` should be current group's size (e.g.: [[0, 1], [2, 3]], `len(x)` should be 2)